Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h

```diff
@@ -0,0 +1,872 @@
+  { }
+
+  void initialize(const std::array<typename Intersection_kernel::Point_3, 8>& bbox, std::vector<std::size_t>& input_polygons) {
+    Timer timer;
+    timer.reset();
+    timer.start();
+
+    std::vector< std::vector<typename Intersection_kernel::Point_3> > bbox_faces;
+    bounding_box_to_polygons(bbox, bbox_faces);
+    add_polygons(bbox_faces, input_polygons);
+
+    m_data.igraph().finished_bbox();
+
+    if (m_parameters.verbose)
+      std::cout << "* intersecting input polygons ... ";
+
+    // Fills in the ivertices on support plane intersections inside the bbox.
+    make_polygons_intersection_free();
+
+    // Generation of ifaces
+    create_ifaces();
+
+    // Splitting the input polygons along intersection lines.
+    initial_polygon_iedge_intersections();
+
+    create_bbox_meshes();
+
+    // Starting from here the intersection graph is const, it won't change anymore.
+    if (m_parameters.verbose)
+      std::cout << "done" << std::endl;
+
+    if (m_parameters.debug)
+      KSP_3::internal::dump(m_data, m_data.prefix() + "intersected");
+
+    CGAL_assertion(m_data.check_bbox());
+    //m_data.set_limit_lines();
+    m_data.precompute_iedge_data();
+
+    m_data.initialization_done();
+
+    if (m_parameters.debug) {
+      for (std::size_t sp = 0; sp < m_data.number_of_support_planes(); sp++)
+        dump_2d_surface_mesh(m_data, sp, m_data.prefix() + "before-partition-sp" + std::to_string(sp));
+    }
+  }
+
+  void clear() {
+    // to be added
+  }
+
+private:
+  std::vector<std::vector<Point_3> >& m_input_polygons;
+  std::vector<typename Intersection_kernel::Plane_3>& m_input_planes;
+  Data_structure& m_data;
+  const Parameters& m_parameters;
+
+  void add_iface_from_iedge(std::size_t sp_idx, IEdge edge, IEdge next, bool cw) {
+    IVertex s = m_data.source(edge);
+    IVertex t = m_data.target(edge);
+
+    IFace face_idx = m_data.add_iface(sp_idx);
+    Face_property& face = m_data.igraph().face(face_idx);
+    face.pts.push_back(m_data.support_plane(sp_idx).to_2d(m_data.igraph().point_3(s)));
+    face.pts.push_back(m_data.support_plane(sp_idx).to_2d(m_data.igraph().point_3(t)));
+    face.vertices.push_back(s);
+    face.vertices.push_back(t);
+    face.edges.push_back(edge);
+    m_data.igraph().add_face(sp_idx, edge, face_idx);
+
+    face.edges.push_back(next);
+    m_data.igraph().add_face(sp_idx, next, face_idx);
+
+    std::size_t iterations = 0;
+
+    int dir = (cw) ? -1 : 1;
+
+    std::size_t inext;
+    while (s != m_data.target(next) && iterations < 10000) {
+      face.vertices.push_back(m_data.target(next));
+      face.pts.push_back(m_data.support_plane(sp_idx).to_2d(m_data.igraph().point_3(m_data.target(next))));
+
+      IEdge enext, eprev;
+      get_prev_next(sp_idx, next, eprev, enext);
+
+      std::vector<std::pair<IEdge, Direction_2> > connected;
+      m_data.get_and_sort_all_connected_iedges(sp_idx, m_data.target(next), connected);
+      inext = -1;
```

Comment:
@sloriot   These `-1`   leads to warnings in some testsuites. Further below we see `static_cast<std::size_t>(-1)`.
Would it make sense to add a variable `null_index` or similar?